### PR TITLE
test: enable integration tests against real S3 (fixes #1899)

### LINKED ISSUE #1899: test: enable integration tests against real S3
# Feature Request

Integration based on AbstractS3Test are assuming `http://localhost:9000` (provided by minio) as endpoint. It would be nice if we can run integration tests against real S3 endpoint based on test time configuration.

## Which Areas Would Be Affected?

test, S3

## Why Is the Feature Desired?

bugs related to specific S3 settings/regions could be found by running tests against real S3.

## Solution Proposal

switching endpoint URI based on the environment variables.

## Type of Issue

improvement

## Checklist

- [x] assigned appropriate label?
- [x] **Do NOT select a milestone or an assignee!**

### PR DESCRIPTION
**Pull request by iwasakims**
## What this PR changes/adds

Make S3 endpoint configurable via system properties.

## Why it does that

For running integration tests against real S3 instead of Minio.

## Further notes

I ran tests against Tokyo region with the following command lines using my AWS credential.

```
$ ./gradlew -p extensions/control-plane/provision/s3-provision clean test -DincludeTags="AwsS3IntegrationTest" --tests '*S3StatusCheckerIntegrationTest' -Dit.aws.profile=iwasakims -Dit.aws.endpoint=https://s3.ap-northeast-1.amazonaws.com/ -Dit.aws.region=ap-northeast-1

> Task :extensions:control-plane:provision:s3-provision:test

---------------------------------------------------------------
|  Results: SUCCESS (6 tests, 6 passed, 0 failed, 0 skipped)  |
---------------------------------------------------------------
```

and.

```
$ ./gradlew -p extensions/data-plane/data-plane-s3 clean test -DincludeTags="IntegrationTest" --tests '*S3DataPlaneIntegrationTest' -Dit.aws.profile=iwasakims -Dit.aws.endpoint=https://s3.ap-northeast-1.amazonaws.com/ -Dit.aws.region=ap-northeast-1

> Task :extensions:data-plane:data-plane-s3:test

---------------------------------------------------------------
|  Results: SUCCESS (1 tests, 1 passed, 0 failed, 0 skipped)  |
---------------------------------------------------------------
```

## Linked Issue(s)

Closes #1899

## Checklist

- [n/a] added appropriate tests?
- [x] performed checkstyle check locally?
- [x] added/updated copyright headers?
- [n/a] documented public classes/methods?
- [x] added/updated relevant documentation?
- [x] assigned appropriate label? (exclude from changelog with label `no-changelog`)
- [x] formatted title correctly? (_take a look at the [CONTRIBUTING](https://github.com/eclipse-dataspaceconnector/DataSpaceConnector/blob/main/CONTRIBUTING.md#submit-a-pull-request) and [styleguide](https://github.com/eclipse-dataspaceconnector/DataSpaceConnector/blob/main/styleguide.md) for details_)

### CODE REVIEW

File: extensions/common/aws/aws-test/src/testFixtures/java/org/eclipse/dataspaceconnector/aws/testfixtures/AbstractS3Test.java [hunk 1]
```diff
@@ -158,7 +166,7 @@ protected void putStringOnBucket(String bucketName, String key, String content)
     }
 
     protected @NotNull AwsCredentials getCredentials() {
-        String profile = propOrEnv("AWS_PROFILE", null);
+        String profile = propOrEnv("it.aws.profile", null);
```
Comment (ndr-brt) [inline extensions/common/aws/aws-test/src/testFixtures/java/org/eclipse/dataspaceconnector/aws/testfixtures/AbstractS3Test.java:169]: please use `var`

**Comment by paullatzelsperger** [comment]
before I review this, please fix the failing tests.

**Comment by iwasakims** [comment]
@paullatzelsperger The cause seems to be running S3DataPlaneIntegrationTest without starting Minio in unexpected jobs like Postgresql-Integration-Tests and Hashicorp-Vault-Integration-Tests. I expect #1897 to address this.

Let me rebase this PR and address review comment of @ndr-brt.

**Comment by iwasakims** [comment]
The cause of the test failure is change of build.gradle.kts overwriting test task setting for skipping IntegrationTests. I will revert the change of build.gradle.kts. We can use environment variable instead of system properties without fixing build.gradle.kts.

```
$ IT_AWS_PROFILE=iwasakims IT_AWS_ENDPOINT=https://s3.ap-northeast-1.amazonaws.com/ IT_AWS_REGION=ap-northeast-1 ./gradlew -p extensions/data-plane/data-plane-s3 clean test -DincludeTags="IntegrationTest" --tests '*S3DataPlaneIntegrationTest' 
```

File: extensions/common/aws/aws-test/src/testFixtures/java/org/eclipse/dataspaceconnector/aws/testfixtures/AbstractS3Test.java (hunk 1 above)
Comment (iwasakims) [reply extensions/common/aws/aws-test/src/testFixtures/java/org/eclipse/dataspaceconnector/aws/testfixtures/AbstractS3Test.java]: done.

File: extensions/common/aws/aws-test/src/testFixtures/java/org/eclipse/dataspaceconnector/aws/testfixtures/AbstractS3Test.java [hunk 2]
```diff
@@ -86,12 +87,19 @@ static void prepareAll() {
...
+    private static boolean isMinio() {
+        return MINIO_ENDPOINT.equals(S3_ENDPOINT.toString());
+    }
+
     /**
      * pings <a href="https://docs.min.io/minio/baremetal/monitoring/healthcheck-probe.html">MinIO's health endpoint</a>
      *
      * @return true if HTTP status [200..300[
      */
     private static boolean pingMinio() throws IOException {
```
[the remark this answers is not in the document]
Comment (iwasakims) [reply extensions/common/aws/aws-test/src/testFixtures/java/org/eclipse/dataspaceconnector/aws/testfixtures/AbstractS3Test.java]: I updated the code based on your suggestion. I chose `isMinioAvailable` preferring `is*` for the name of method returning boolean. @ndr-brt

File: extensions/common/aws/aws-test/src/testFixtures/java/org/eclipse/dataspaceconnector/aws/testfixtures/AbstractS3Test.java [hunk 3]
```diff
@@ -83,15 +85,25 @@ static void prepareAll() {
                 .pollInterval(Duration.ofSeconds(2))
                 .ignoreException(IOException.class) // thrown by pingMinio
                 .ignoreException(ConnectException.class)
-                .until(AbstractS3Test::pingMinio);
+                .until(() -> {
```
Comment (paullatzelsperger) [inline extensions/common/aws/aws-test/src/testFixtures/java/org/eclipse/dataspaceconnector/aws/testfixtures/AbstractS3Test.java:88]: minor nit: introducing a method, something like `isBackendAvailable` or similar naming would make this piece of code more readable:
```
.ignoreException(ConnectException.class)
.until(this::isBackendAvailable)

// and then:
private boolean isBackendAvailable(){
   if(isMinio()){
      return isMinioAvailable();
   } else {
      return true;
   }
};
```

Comment (iwasakims) [reply extensions/common/aws/aws-test/src/testFixtures/java/org/eclipse/dataspaceconnector/aws/testfixtures/AbstractS3Test.java]: Sure. I updated the code.

**Review by ndr-brt — APPROVED** [review]

**Review by paullatzelsperger — APPROVED** [review]

**Pull request merged**